File: 82. 删除排序链表中的重复元素 II/3.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if (!head || !head->next) {
            return head;
        }

        if (head->next->val == head->val) {
            while (head->next && head->next->val == head->val) {
                head->next = head->next->next;
            }

            head = deleteDuplicates(head->next);
        } else {
            head->next = deleteDuplicates(head->next);
        }

        return head;
    }
};

File: 206. 反转链表/3.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	return reverse(nil, head)
}

func reverse(previous, current *ListNode) *ListNode {
	if current == nil {
		return previous
	}

	tmp := current.Next
	current.Next = previous
	return reverse(current, tmp)
}